### ISSUE TITLE
ban rule not working

### ISSUE THREAD

**Issue by lavie** [palantir/tslint#3893]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7.2
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
import * as uuid from 'uuid';
uuid.v1();
```

with `tslint.json` configuration:

```json
"rules": {
  "ban": ["v1", ["uuid", "v1"]]
}
```

#### Actual behavior

No lint errors.

#### Expected behavior

The call to `uuid.v1` should have been banned.

My apologies if I'm simply misusing the rule somehow. I tried it in various configuration variations.

```yaml
rules:
  ban:
    - "v1"
    - ["v1"]
    - name:
        - uuid
        - v1
```

**Comment by bowenni** [comment]
Looking at https://palantir.github.io/tslint/rules/ban/, I think this configuration should trigger the expected lint error:
```json
{"rules":{"ban":[true,{"name":["uuid","v1"]}]}}
```
